### ISSUE TITLE
Creating a filter generates duplicate events in other filters

### ISSUE THREAD

**Issue by pcppcp** [ethereum/eth-tester#112]
* Version: 0.1.0b27
* Python: 3.5
* OS: linux

### What was wrong?

When creating a new filter using `create_log_filter()`, `_process_blocks_logs()` is called to populate filter with past events.
This method however calls `add()` method for all existing filters:
```python
    def _process_block_logs(self, block):
        for _, filter in self._log_filters.items():
            for transaction_hash in block['transactions']:
                receipt = self.get_transaction_receipt(transaction_hash)
                for log_entry in receipt['logs']:
                    raw_log_entry = self.normalizer.normalize_inbound_log_entry(log_entry)
                    filter.add(raw_log_entry)
```

### How can it be fixed?

`Filter::add()` method should check if the log entry already exists in `Filter::values` before appending/processing it.

**Comment by pcppcp** [comment]
Might be also related to this `web3` issue

https://github.com/ethereum/web3.py/issues/907

**Comment by carver** [comment]
Closed by #113
